### PR TITLE
Add `WindowConfig` for `OpenWindow`

### PR DESCRIPTION
**Pull request by zchee**
Add `WindowConfig` struct for `OpenWindow` API instead of `map[string]interface`.
Also, added `floating_window` testcase.

Close: #54

### CODE REVIEW

File: nvim/types.go [hunk 1]
```diff
@@ -235,3 +235,60 @@ type VirtualTextChunk struct {
...
+//
+// Style is the Configure the appearance of the window.
+// Currently only takes one non-empty value:
+//
+//  minimal:
+//    Nvim will display the window with many UI options disabled.
+//    This is useful when displaying a temporary float where the text should not be edited.
+//    Disables 'number', 'relativenumber', 'cursorline', 'cursorcolumn','foldcolumn', 'spell' and 'list' options. 'signcolumn' is changed to `auto`.
+//    The end-of-buffer region is hidden by setting `eob` flag of 'fillchars' to a space char, and clearing the EndOfBuffer region in 'winhighlight'.
+type OpenWindowConfig struct {
```
Comment (justinmk) [inline nvim/types.go]: why name it `OpenWindowConfig` instead of just `WindowConfig` ?

**Comment by justinmk** [comment]
> @justinmk did you enjoy VimConf in Japan? :D

yes it was a good experience! were you there? :)

File: nvim/types.go (hunk 1 above)
Comment (zchee) [reply nvim/types.go]: Just renamed.

**Comment by zchee** [comment]
> yes it was a good experience! were you there? :)

No :( I really went to go VimConf...
But I am happy to hear that :D Also, your slide is very good works, interesting!!

BTW, PTAL

File: nvim/nvim_test.go [hunk 2]
```diff
@@ -515,6 +515,93 @@ func TestAPI(t *testing.T) {
...
+		if cursorlineOpt {
+			t.Fatal("style is minimal; expected the cursorline window option to disabled")
+		}
+		if cursorcolumnOpt {
+			t.Fatal("style is minimal; expected the cursorcolumn window option to disabled")
+		}
+		if spellOpt {
+			t.Fatal("style is minimal; expected the spell window option to disabled")
+		}
+		if listOpt {
```
Comment (justinmk) [inline nvim/nvim_test.go]: this is pretty verbose. Maybe next time just check all options at once, and have 1 generic message.

```
if numberOpt || relativenumberOpt || cursorlineOpt || ... {
  t.Fatal("expected minimal style")
}
```

**Comment by zchee** [comment]
@justinmk Thanks re-approve! will merge

**Comment by zchee** [comment]
@justinmk Also, I was remove support old go versions. will cut `v1.0.1` release.

**Comment by justinmk** [comment]
> remove support old go versions

@akiyosi @dzhou121 any objection?

**Comment by zchee** [comment]
@justinmk dzhou121's popular repo which used neovim/go-client is here: (I don't know he have another repo used go-client)

https://github.com/dzhou121/gonvim

But IIRC has been not maintain this repo. (Ibrhink you already know)
And @akiyosi forked this. also very active.

So, I think would be good to adding @akiyosi to @neovim/golang member if we will get reply cooment for

> any objection?

also  if he want to it.

---

BTW, I saw @neovim/golang team members role.
Unfortunately, now have `maintainer` label users not maintain go-client repo, (but yes, I really respect great works. made any SDK from scratch is very hard work) that's same means of there is no go-client maintainer.
So, could you change my role to `Maintainer`? if haven't any concerns such as neovim org policy, GitHub org permission scope or etc.

**Comment by akiyosi** [comment]
Hi
Sorry for late reaction.

> remove support old go versions

I have no objection.

> So, I think would be good to adding @akiyosi to @neovim/golang member if we will get reply cooment for
> also if he want to it.

If I can help the project even a little, I would like to join the members.

**Pull request merged**